sway/input/seat.c

```diff
@@ -73,6 +73,24 @@ static void seat_node_destroy(struct sway_seat_node *seat_node) {
 	free(seat_node);
 }
 
+void seat_notify_key_press(struct sway_seat *seat) {
+	if (seat->cursor->hidden) {
+		return;
+	}
+
+	const struct seat_config *seat_config = seat_get_config(seat);
```

Comment:
Looking up the config for every keypress seems excessive, especially as it happens twice in the worst case where the config is for `*` and hide when typing isn't enabled,

Cache the value on the seat or the cursor, so the function just becomes something like:
```
if (!seat->cursor->hidden && seat->cursor->hide_when_typing) {
    cursor_hide(seat->cursor);
}
```

Reply:
I agree that it's excessive. The issue of course with having a cache is that you need to invalidate it whenever the configuration changes. Right now I'm thinking to just go over all the seats and invalidate their cache whenever a `hide_cursor_when_typing` configuration changes. Not the prettiest solution but I'm not sure there is a better one.